### DISCUSSION TITLE
Strongswam openssl integration

### DISCUSSION THREAD

**Discussion by alireza11048** [strongswan/strongswan#510] [Q&A] [answered]
Hi to all,

I am trying to integrate a hardware SSL accelerator to the strongswan, to test how it affects the performance of strongswan. my hardware accelerator can be accessed through an OpenSSL engine. is there any way to tell strongswan to use openssl with a specified engine to do cryptographic primitives? 

for example i can use my hardware accelerator with the below command:
/open_ssl_address/openssl <cmd> -engine cavium <cmd_parameter>

Thanks,

**Comment by Thermi** [comment, marked as answer]
please don't cross post.

Am July 17, 2021 5:26:40 PM UTC schrieb alireza11048 ***@***.***>:
>Hi to all,
>
>I am trying to integrate a hardware SSL accelerator to the strongswan,
>to test how it affects the performance of strongswam. my hardware
>accelerator can be accessed through an OpenSSL engine. is there any way
>to tell strongswam to use openssl with a specified engine to do
>cryptographic primitives? 
>
>for example i can use my hardware accelerator with the below command:
>/open_ssl_address/openssl <cmd> -engine cavium <cmd_parameter>
>
>Thanks,
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/strongswan/strongswan/discussions/510

-- 
Sent from mobile

**Reply by Thermi** [reply]
For strongswan you can use the `charon.plugins.openssl.engine_id` key in strongswan.conf.
For the kernel you need a kernel crypto driver.

**Reply by alireza11048** [reply]
Thanks for your advice,
I have a kernel driver to communicate with the hardware accelerator, and as I said I can easily access it with the OpenSSL engine option.

I have a default OpenSSL on my os, and another in /usr/local/ssl folder (which contains headerfiles, static and dynamic library files of OpenSSL) that supports my hardware accelerator.
Could you please tell me that how can I compile the StrongSwan with the OpenSSL which is located in the "/usr/local/ssl" instead of the system default one. and also use the libcrypto.a or libcrypto.so which is located in /usr/local/ssl?

In https://wiki.strongswan.org/issues/2517 I've found something related, but I could not able to crack this thing.

Thank you

**Reply by Thermi** [reply]
Just pass the necessary flags using CFLAGS at build time.
